src/services/entity_repository.js

```diff
@@ -135,7 +102,49 @@ export default class EntityRepository {
     await this.updateEntities(entities, update);
   }
 
-  async endBundle(bundleStubId, bundleId, bundleSizeLimit) {
+  async fetchEntitiesForBundling(bundleStubId, bundleSizeLimit) {
+    const notBundledQuery = {
+      'metadata.bundleId': null,
+      'repository.bundleStubId': null
+    };
+
+    const setBundleStubIdUpdate = {
+      $set: {
+        'repository.bundleStubId': bundleStubId
+      }
+    };
+
+    await this.db.collection('assets').update(notBundledQuery, setBundleStubIdUpdate, {multi: true});
+    await this.db.collection('events').update(notBundledQuery, setBundleStubIdUpdate, {multi: true});
+
+    const thisBundleStubQuery = {
+      'repository.bundleStubId': bundleStubId
+    };
+
+    let assets = await this.db
+      .collection('assets')
+      .find(thisBundleStubQuery, {fields: this.blacklistedFields})
+      .toArray();
+    let events = await this.db
+      .collection('events')
+      .find(thisBundleStubQuery, {fields: this.blacklistedFields})
+      .toArray();
+
+
+    const entities = this.orderedEntityIds(assets, events);
+    const discardedEntities = entities.slice(bundleSizeLimit);
+    await this.unsetEntitiesBundlesStubs(discardedEntities);
+
+    assets = assets.filter((asset) => !discardedEntities.some((id) => asset.assetId === id));
+    events = events.filter((event) => !discardedEntities.some((id) => event.eventId === id));
```

Comment:
Why not `discardedEntities.includes(asset.assetId)`. We can optimize it even more by using Set

Reply:
Propose exact implementation, because I'm not sure if we can really do it better.